lib/CamDiag.py

```diff
@@ -456,18 +506,8 @@ def regrid_climo(self):
                 msg = "Regridding file '{}' is missing. Script is ending here.".format(regrid_script_path)
                 end_diag_script(msg)
 
-            #Create regridding script import statement:
-            regrid_func_import_statement = "from {} import {}".format(regrid_func_name, regrid_func_name)
-
-            #Run regridding script import statement:
-            exec(regrid_func_import_statement)
-
-            #Create actual function call:
-            regrid_func = regrid_func_name+\
-            '(case_name, input_climo_loc, output_loc, var_list, target_list, target_loc, overwrite_regrid)'
-
-            #Evaluate (run) averaging script function:
-            eval(regrid_func)
+            regrid_func_args = [case_name, input_climo_loc, output_loc, var_list, target_list, target_loc, overwrite_regrid]
+            function_caller(regrid_func_name, regrid_func_args, func_kwargs={}, module_name=regrid_func_name+'.py')
```

Comment:
It might make sense to add all of the extra `opt` code you have in `create_climo` here as well.  If the code works as expected then it shouldn't modify the current behavior, but will allow one to add arguments to regridding scripts in the future.